.github/copilot-instructions.md

```diff
@@ -0,0 +1,233 @@
+- `https://artifacts.camunda.com/artifactory/camunda-bpm-snapshots/` for Camunda BPM snapshots
+
+No additional settings.xml configuration is required.
+
+### Building the Project
+
+```bash
+# Full build with all tests
+mvn clean install
+
+# Build without tests (faster for iteration)
+mvn clean install -DskipTests
+
+# Build excluding Cockpit plugin (requires Node.js)
+mvn clean install -pl '!plugins/cockpit'
+```
+
+### Running Tests
+
+```bash
+# All tests
+mvn verify
+
+# Unit tests only
+mvn test
+
+# Integration tests only
+mvn integration-test
+
+# Tests with specific database
+mvn verify -Ppostgresql
+mvn verify -Poracle
+```
+
+### Frontend Development (Cockpit Plugin)
+
+```bash
+cd plugins/cockpit/frontend
+
+# Install dependencies
+npm install
+
+# Development build with watch
+npm run dev
+
+# Production build
+npm run build
+```
+
+## Development Workflow
+
+### Prerequisites
+
+- Java 21 must be installed and set as JAVA_HOME
+- Maven 3.6+ must be available
+- Node.js is required only for the Cockpit plugin
+- Docker is recommended for database testing
+
+### Making Changes
+
+1. **Module-Specific Builds:** You can build individual modules by running Maven commands in their directories to speed up development
+2. **Incremental Testing:** Run tests relevant to your changes before running the full test suite
+3. **Database Testing:** Use Docker to test with PostgreSQL or Oracle if making database-related changes
+4. **Distribution Testing:** After significant changes, build the full distribution from `assembly/target/`
+
+### Code Changes Guidelines
+
+- Make minimal, focused changes that address the specific issue
+- Don't modify working code unnecessarily
+- Update documentation when adding or changing features:
+  - For complex changes, add comments in the code explaining the approach
+  - Update JavaDoc for all public API changes
+  - Update README.md if the changes affect installation, usage, or features
+  - If changes require updates to the official documentation at docs.camunda.io, note this in the PR description
+- Ensure changes are backward compatible unless explicitly intended otherwise
+- Be mindful of production readiness status (runtime vs. history migrator)
+
+## Common Patterns
+
+### Variable Interceptors
+
+When creating or modifying variable interceptors:
+- Extend the `VariableInterceptor` interface
+- Package as standalone JAR when needed
+- See `examples/variable-interceptor/` for reference implementation
+
+### Database Support
+
+When adding database-specific code:
+- Test with H2, PostgreSQL, and Oracle profiles
+- Use JDBC standards to maintain compatibility
+- Add profile-specific tests in the `qa/` module
+
+### Spring Boot Configuration
+
+- Use `application.properties` or `application.yml` for configuration
+- Support externalized configuration for deployment flexibility
+- Document all configuration properties
+
+## Pull Request Quality
+
+### Commit Messages
```

Comment:
We can add that we usually reference the issue number in the commit message body: `related to #<issue-number>`